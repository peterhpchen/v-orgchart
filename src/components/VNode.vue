<template>
  <div class="node">
    <slot>
      <!-- <div>{{nodeData.name}}</div> -->
    </slot>
  </div>
</template>

<script>
export default {
  // props: {
  //   nodeData: {
  //     type: Object,
  //     required: true,
  //   }
  // }
};
</script>

<style>

</style>
