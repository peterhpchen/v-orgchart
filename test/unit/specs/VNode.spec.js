import { shallow } from '@vue/test-utils';
import VNode from '@/components/VNode';

describe('VNode.vue', () => {
  it('match snapshot', () => {
    const node = {
      name: 'A',
    };

    const wrapper = shallow(VNode, {
      propsData: {
        data: node,
      },
      scopedSlots: {
        default: '<div class="name" slot-scope="slotProps">{{ slotProps.data.name }}</div>',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
