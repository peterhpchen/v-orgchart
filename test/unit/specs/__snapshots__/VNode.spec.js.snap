// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VNode.vue match snapshot 1`] = `
<div class="node">
    <div class="name">A</div>
</div>
`;
